pkg/util/kubernetes/apiserver/metadata_controller.go

```diff
@@ -325,6 +325,25 @@ func GetPodMetadataNames(nodeName, ns, podName string) ([]string, error) {
 	for _, s := range serviceList {
 		metaList = append(metaList, fmt.Sprintf("kube_service:%s", s))
 	}
-
 	return metaList, nil
 }
+
+// GetNodeLabels retrieves the labels of the queried node from the cache of the shared informer.
+func GetNodeLabels(nodeName string) (map[string]string, error) {
+	as, err := GetAPIClient()
+	if err != nil {
+		return nil, err
+	}
+	enabled := config.Datadog.GetBool("kubernetes_collect_metadata_tags")
+	if !enabled {
```

Comment:
I feel `enabled` isn't really necessary there

Reply:
it is, as we look into the informer's cache. This condition is required to start the metadata controller which instantiates the cache,. https://github.com/DataDog/datadog-agent/blob/master/pkg/util/kubernetes/apiserver/controllers.go#L26

Reply:
I was more meaning `if config.Datadog.GetBool("kubernetes_collect_metadata_tags")` 😬

Reply:
ahhh 🤦‍♂️